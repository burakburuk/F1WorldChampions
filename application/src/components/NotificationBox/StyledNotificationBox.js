import styled from 'styled-components';

const StyledNotificationBox = styled.div`
    position: fixed; /* Stay in place */
    z-index: 2; /* Sit on top */
    right: 10px;
    bottom: 10px;
    min-height: 50px;
    overflow: hidden;
    color: #D8000C;
    background-color: #FFBABA;
    padding: 12px 30px;
    display: ${props => (props.open ? 'block' : 'none')};
`;

export default StyledNotificationBox;
