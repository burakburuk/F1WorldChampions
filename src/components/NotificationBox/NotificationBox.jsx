import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledNotificationBox = styled.div`
    position: fixed; /* Stay in place */
    z-index: 2; /* Sit on top */
    right: 10px;
    bottom: 10px;
    min-height: 50px;
    overflow: hidden;
    color: #D8000C;
    background-color: #FFBABA;
    padding: 12px 30px;
    display: ${props => (props.open ? 'block' : 'none')};
`;

const StyledHeader = styled.div`

    .close {
        float: right;
        font-size: 22px;
        font-weight: bold;
        width: 20px;
        height: 20px;
        background: red;
        color: #FFBABA;
        text-align: center;
        line-height: 22px;
        margin-right: -22px;
        margin-top: -5px;
    }

    .close:hover,
    .close:focus {
        text-decoration: none;
        cursor: pointer;
    }
`;

const NotificationBox = props => (
    <StyledNotificationBox
        open={props.open}>
        <StyledHeader>
            <span
                className='close'
                onClick={props.onClose}>
                &times;
            </span>
            <h2>{props.title}</h2>
        </StyledHeader>
        {props.children}
    </StyledNotificationBox>
);

NotificationBox.propTypes = {
    children: PropTypes.string,
    open: PropTypes.bool.isRequired,
    title: PropTypes.string,
    onClose: PropTypes.func,
};

export default NotificationBox;
