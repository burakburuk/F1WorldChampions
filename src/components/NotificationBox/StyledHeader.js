import styled from 'styled-components';

const StyledHeader = styled.div`

    .close {
        float: right;
        font-size: 22px;
        font-weight: bold;
        width: 20px;
        height: 20px;
        background: red;
        color: #FFBABA;
        text-align: center;
        line-height: 22px;
        margin-right: -22px;
        margin-top: -5px;
    }

    .close:hover,
    .close:focus {
        text-decoration: none;
        cursor: pointer;
    }
`;

export default StyledHeader;
