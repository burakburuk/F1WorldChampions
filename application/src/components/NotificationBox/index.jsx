import React from 'react';
import PropTypes from 'prop-types';
import StyledNotificationBox from './StyledNotificationBox';
import StyledHeader from './StyledHeader';

const NotificationBox = props => (
    <StyledNotificationBox
        data-id="notification"
        open={props.open}>
        <StyledHeader>
            <span
                className='close'
                onClick={props.onClose}>
                &times;
            </span>
            <h2>{props.title}</h2>
        </StyledHeader>
        {props.children}
    </StyledNotificationBox>
);

NotificationBox.propTypes = {
    children: PropTypes.string,
    open: PropTypes.bool.isRequired,
    title: PropTypes.string,
    onClose: PropTypes.func,
};

export default NotificationBox;
