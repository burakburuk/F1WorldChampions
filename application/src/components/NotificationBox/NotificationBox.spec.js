import React from 'react';
import { shallow } from 'enzyme';
import NotificationBox from './';

it('should match with snapshot', () => {
    const wrapper = shallow(
        <NotificationBox
            open={true}
            title="test title"
        />
    );
    expect(wrapper).toMatchSnapshot();
});
