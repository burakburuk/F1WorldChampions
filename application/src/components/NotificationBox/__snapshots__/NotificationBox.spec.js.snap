// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<styled.div
  data-id="notification"
  open={true}
>
  <styled.div>
    <span
      className="close"
    >
      Ã—
    </span>
    <h2>
      test title
    </h2>
  </styled.div>
</styled.div>
`;
